### DISCUSSION TITLE
Accessing request context within SecuritySchemas method

### DISCUSSION THREAD

**Discussion by sugap91** [spec-first/connexion#1836] [Q&A]
We are in the midst of upgrading connexion 2.x to 3.x and using it within Azure function apps (serverless). Converting FlaskApp to AsyncApp.
```
app = connexion.AsyncApp(__name__)
app.add_api(specification)
await func.AsgiMiddleware(app).handle_async(event, context)
```

We are using securitySchemes (x-bearerInfoFunc) in openapi for authentication and within the method we would like to access request context in order to access request method and url path to do some authorisation checks.

We were able to access it in connexion 2.x by `flask.request.method` and `flask.request.path`.
However, I am unable to access it in connexion 3.0 using `from connexion import request` and ` request.context`. Facing the below error while trying to access,
`Working outside of operation context. Make sure your app is wrapped in a ContextMiddleware and you're processing a request while accessing the context.`

Only the bearer token gets passed to the securitySchemes method by default. Is there a way to access request variables within the method?

**Comment by chrisinmtown** [comment]
I think a reply that's exactly one year late is not helpful, but I'll post it anyhow. I hit a similar problem yesterday, only difference is that I'm using BasicAuth (not bearerInfo).  Nope, the `request` is no longer available as a kind-of global.  Happily the Connexion doc https://connexion.readthedocs.io/en/stable/security.html explains fairly clearly that you can declare parameter `request` in your auth-handler function and the request object will be injected. Took me a couple hours to find that.

Also see issue https://github.com/spec-first/connexion/issues/2015

p.s. you might want to close this issue if you have it working
